### DISCUSSION TITLE
Custom Length Units

### DISCUSSION THREAD

**Discussion by GiacomoZheng** [typst/typst#3172] [Q&A] [answered]
Is it possible to define a new length unit? For example, koli-pt, and I want to write something like `5kpt` or similar.

I found a similar question in Tex: https://tex.stackexchange.com/q/343255 .

**Comment by frozolotl** [comment, marked as answer]
That is not possible, but you can do the same with a simple function:
```typ
#let kpt(n) = n * 2pt
#rect(width: kpt(50), height: kpt(10))
```

Or alternatively using multiplication, if you prefer:
```typ
#let kpt = 2pt
#rect(width: 50 * kpt, height: 10 * kpt)
```

**Reply by GiacomoZheng** [reply]
Thank you for your reply. Do you know it is a good idea to make a "feature request" issue?

**Reply by frozolotl** [reply]
Well, it doesn't hurt to create one, but I'm not sure that feature would be accepted, unless there is a good reason for it.
It might look differently if it is about one specific length unit.

**Reply by GiacomoZheng** [reply]
I thought about it again, and I agree it is useless for me.
Maybe it will be useful for compatibility? Since some countries have their own units.
